File: Basic of CP/Sum of Digits/Sum of Digit.py
def getSum(n):
 
    sum = 0
    while (n != 0):
 
        sum = sum + int(n % 10)
        n = int(n/10)
 
    return sum
 
n = 687
print(getSum(n))

### Recursive Problem
def sumDigits(no):
    return 0 if no == 0 else int(no % 10) + sumDigits(int(no/10))
 
print(sumDigits(68724))


### Taking input as a string
def getSum(n):
    
    sum = 0       # Initializing sum to 0
    
    for i in n:     # Traversing through string
        
        sum = sum + int(i)   # Converting char to int
 
    return sum

n = "123456789123456789123422"
print(getSum(n))

File: Basic of CP/Sum of Digits/Sum of Digit.cpp
// C program to compute sum of digits in
// number.
#include <iostream>
using namespace std;

/* Function to get sum of digits */
class gfg {
public:
	int getSum(int n)
	{
		int sum = 0;
		while (n != 0) {
			sum = sum + n % 10;
			n = n / 10;
		}
		return sum;
	}
};

// Driver code
int main()
{
	gfg g;
	int n = 687;
	cout << g.getSum(n);
	return 0;
}
// This code is contributed by Soumik


// C++ implementation of the above approach
#include <iostream>
using namespace std;
int getSum(string str)
{
	int sum = 0;

	// Traversing through the string
	for (int i = 0; i < str.length(); i++) {
		// Since ascii value of
		// numbers starts from 48
		// so we subtract it from sum
		sum = sum + str[i] - 48;
	}
	return sum;
}

// Driver Code
int main()
{
	string st = "123456789123456789123422";
	cout << getSum(st);
	return 0;
}
